src/mlpack/methods/ann/layer/layer_types.hpp

```diff
@@ -92,6 +109,7 @@ using LayerTypes = boost::variant<
     DropConnect<arma::mat, arma::mat>*,
     Dropout<arma::mat, arma::mat>*,
     ELU<arma::mat, arma::mat>*,
+    FFN<NegativeLogLikelihood<>, RandomInitialization>*,
```

Comment:
@partobs-mdp: I think this comment is relevant to the comment you made on this PR too---if you need to add networks to internal layers that you're building (I think you have to for HAM), this could be another approach that could work for that and not have the drawback you pointed out.